code-challenges.test.js

```diff
@@ -55,6 +102,34 @@ var numbersToAdd2 = [0, 7, -8, 12]
 var numbersToAdd3 = []
 // Expected output: []
 
+describe("accumulationStation", () => {
+  it("takes in an array and returns an array of the accumulating sum", () =>{
+      expect(accumulationStation(numbersToAdd1)).toEqual([2, 6, 51, 60])
+      expect(accumulationStation(numbersToAdd2)).toEqual([0, 7, -1, 11])
+      expect(accumulationStation(numbersToAdd3)).toEqual([])
+  })
+})
 
+//Test failed successfully
 
 // b) Create the function that makes the test pass.
+
+//I'll want to declare a function with an array as an input
+const accumulationStation = (arr) => {
+  //I'll need an empty array to push new values onto
+  let newArr = []
+  //Array iteration to perform code on each element in the array
+  for (i=0; i<arr.length; i++){
+    //I only want to push code onto values that sit at a higher index than 0/first on the array, otherwise the code will return NaN. A condition of i>0 will only iterate the and push the code on values on indexes above the 0/first array index. 
+    if (i>0){
+      //I want to push the sum of the current value at the current index and the accumulating sum of previous values of the previous indexes
+      newArr.push(arr[i] += arr[i-1])
+      //An else statement will push the original value at the 0/first index of the original array onto the new array in the first place holder, because it will always begin the iteration at index 0. 
+      } else {
+        newArr.push(arr[0])
```

Comment:
This works out well! Feel free to brainstorm in comments how you might refactor the problem. You don't have to follow up on it but it's a helpful exercise.